### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by yuce**

### CODE REVIEW

File: docs/modules/ROOT/pages/overview.adoc [hunk 1]
```diff
@@ -32,31 +32,32 @@ Get xref:clc-map.adoc[direct access to map data] for quick debugging of prototyp
 
 === Scripting for Automation
 
-Work in <<non-interactive-mode, non-interactive mode>> to script and execute repetitive administration, integration, or testing tasks.
+Write scripts execute repetitive administration, integration, or testing tasks.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/overview.adoc:35]:
```suggestion
Write scripts to execute repetitive administration, integration, or testing tasks.
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 2]
```diff
@@ -78,7 +79,7 @@ In the interactive mode, you can run SQL queries directly. Make sure to add a se
 ---------------------------------------------------------------------------------
 ----
 
-In order to run CLC commands, prefix them with a backslash `\`:
+In order to run CLC commands, prefix them with a backslash (`\`):
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/overview.adoc:82]:
```suggestion
To run CLC commands, prefix them with a backslash (`\`):
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 3]
```diff
@@ -91,6 +92,47 @@ In order to run CLC commands, prefix them with a backslash `\`:
 ------------------------------------
 ----
 
+You can enter multiline CLC commands by adding a backslash (`\`) character at the very end of every line, except the last one.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/overview.adoc:95]:
```suggestion
You can enter multi-line CLC commands by adding a backslash (`\`) character at the end of every line, except the last one.
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 4]
```diff
@@ -91,6 +92,47 @@ In order to run CLC commands, prefix them with a backslash `\`:
...
+The continuation lines are concatenated, so it may be necessary to put one or more spaces before the backslash:
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/overview.adoc:96]:
```suggestion
The continuation lines are concatenated, so you might need to add one or more spaces before the backslash:
```

File: docs/modules/ROOT/pages/overview.adoc [hunk 5]
```diff
@@ -91,6 +92,47 @@ In order to run CLC commands, prefix them with a backslash `\`:
...
+> \map --name set my-map "key with whitespace" "value with whitespace"
+> \map --name get my-map "key with whitespace"
+-----------------------
+ this
+-----------------------
+ value with whitespace
+-----------------------
+----
+
+Alternatively, you can escape space character with a backslash:
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/overview.adoc:114]:
```suggestion
Alternatively, you can escape the space character with a backslash:
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/advanced-scripting.adoc [hunk 6]
```diff
@@ -385,7 +385,7 @@ print("My home directory:", user_home)
 
 `now()` function returns the current local time in nanoseconds.
 
-You can convert the time to other units using the predefined time units, such as `millisecond`, `second` and `day`.
+You can convert the time to other units using the predefined time units, such as `MILLISECOND`, `SECOND` and `DAY`.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/advanced-scripting.adoc:388]: Thanks Rebekah,
It's in the `clc script run` reference, but I've forgotten to update it before.
IT's now updated:
https://github.com/hazelcast/clc-docs/pull/12/commits/6d7de5131bc88aeb633afb27ebab1d94b1d5490e

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 7]
```diff
@@ -1,6 +1,6 @@
-= clc serializer generator (Beta)
+= clc serializer (Beta)
 
-Serializer commands generate serializers for various serialization systems.
+Generates serializer code for various serialization systems.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc:3]: We currently only support compact serialization, but we'll add more in the future.
I will change it to just "Generates serialization code" if it's OK.

File: docs/modules/ROOT/pages/advanced-scripting.adoc [hunk 8]
```diff
@@ -676,37 +676,37 @@ Whenever an error occurs, CLC stops the script and prints an error message.
 This is usually the safest way to handle the error.
 But there are times when you want to try something that can result in an error, and handle it without giving back the control.
 
-The `script` command supports `the --ignore-errors` flag which suppresses errors while running the script.
+The `script` command supports the `--ignore-errors` flag which suppresses errors while running the script.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/advanced-scripting.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-atomicreference.adoc [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+Example:
+
+[source,bash]
+----
+clc atomicreference --name ref1 get my-value
+----
+
+== clc atomicreference get-and-set
+
+Sets a value in the given AtomicReference and returns the old value.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-atomicreference.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc (hunk 7 above)
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 10]
```diff
@@ -61,83 +61,86 @@ A schema allows you to:
 * Describe the contents of a compact class using supported field types.
 * Import other schema.
 * Specify a namespaces for schema files and reference other namespaces.
-* Define cyclic references between classes.
 * Reference classes that are not present in the given schemas.
 
-A schema is written in YAML. Schema format is given below:
+Schemas are written in YAML.
+
+Schema format is given below:
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 11]
```diff
@@ -61,83 +61,86 @@ A schema allows you to:
...
-*** `nullableFloat32`
-*** `nullableFloat32[]`
-*** `nullableFloat64`
-*** `nullableFloat64[]`
-*** `<OtherCompactClass[]>`
-** external:
-*** Used to mark if the type is external. If a field is external, the tool will not check if it is imported and available. External types are managed by the user and not generated by the tool.
-*** The serializer of an external field can be a custom serializer which is handwritten, the zero-config serializer for Java and .NET, or previously genereated using the tool. This flag will enable such mixed use cases.
-*** In generated code, external types are imported exactly what as the "type" of the field, hence for languages like Java the user should enter the full package name together with the class. E.g. type: `com.app1.dto.Address`.
+Defines class informations in the schema.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 12]
```diff
@@ -61,83 +61,86 @@ A schema allows you to:
...
+
+All class informations have the following attributes:
+
+* `name`: Full name of the class, including package/namespace.
+* `typename`: This is an optional type name to be used. Compact serialization mechanism distinguishes data types in the cluster using their type names. Hence, type names must be unique across the cluster. If not specified, the full type name of the class is used.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 13]
```diff
@@ -61,83 +61,86 @@ A schema allows you to:
...
+* `fields`: Field informations of the class
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 14]
```diff
@@ -61,83 +61,86 @@ A schema allows you to:
...
+
+A field information has the following attributes:
+
+* `name`: Name of the field
+* `type`: EIther one of the builtin type names llsted below, or full name of the field type, including package/namespace.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/configuration.adoc [hunk 15]
```diff
@@ -23,26 +23,22 @@ Known configurations can be directly specified by their names, instead of the fu
...
 pr-3066
+--------------------
 
 # Start CLC shell with configuration named pr-3066
 $ clc -c pr-3066
 ----
 
-If no configuration is specified, the `default` configuration is used.
+. If no configuration is specified, and there is a single configuration, that configuration is used.
+. If there are more than one configurations and the `default` configuration exists, that is used.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/configuration.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/configuration.adoc [hunk 16]
```diff
@@ -23,26 +23,22 @@ Known configurations can be directly specified by their names, instead of the fu
...
 default
 pr-3066
+--------------------
 
 # Start CLC shell with configuration named pr-3066
 $ clc -c pr-3066
 ----
 
-If no configuration is specified, the `default` configuration is used.
+. If no configuration is specified, and there is a single configuration, that configuration is used.
```
[the remark this answers is not in the document]
Comment (yuce) [reply docs/modules/ROOT/pages/configuration.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/overview.adoc (hunk 2 above)
Comment (yuce) [reply docs/modules/ROOT/pages/overview.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/overview.adoc (hunk 3 above)
Comment (yuce) [reply docs/modules/ROOT/pages/overview.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/overview.adoc (hunk 4 above)
Comment (yuce) [reply docs/modules/ROOT/pages/overview.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/overview.adoc (hunk 5 above)
Comment (yuce) [reply docs/modules/ROOT/pages/overview.adoc]: FIxed at: https://github.com/hazelcast/clc-docs/pull/12/commits/2abc5d34626d8e6a8e69250ca107f077c31bd8df

File: docs/modules/ROOT/pages/clc-serializer.adoc [hunk 17]
```diff
@@ -88,18 +88,18 @@ The options apply to all serializer classes generated from the classes in this f
...
-* `typename`: This is an optional type name to be used. Compact serialization mechanism distinguishes data types in the cluster using their type names. Hence, type names must be unique across the cluster. If not specified, the full type name of the class is used.
-* `fields`: Field informations of the class
+* `typename`: This is an optional type name to be used. The compact serialization mechanism distinguishes data types in the cluster using their type names. This means that type names must be unique across the cluster. If not specified, the full type name of the class is used.
+* `fields`: Field information of the class
 
 A field information has the following attributes:
 
 * `name`: Name of the field
-* `type`: EIther one of the builtin type names llsted below, or full name of the field type, including package/namespace.
+* `type`: Either one of the built0in type names listed below, or the full name of the field type, including the package/namespace.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/clc-serializer.adoc:102]:
```suggestion
* `type`: Either one of the built-in type names listed below, or the full name of the field type, including the package/namespace.
```

**Review by rebekah-lawrence — APPROVED** [review]
Minor typo to update, otherwise LGTM

File: docs/modules/ROOT/pages/clc-serializer.adoc (hunk 17 above)
Comment (yuce) [reply docs/modules/ROOT/pages/clc-serializer.adoc]: Oops! Fixed at: https://github.com/hazelcast/clc-docs/pull/12/commits/42fe9beaa1c49bb95652100880bea35baa44869c

**Pull request merged**